mmdeploy/codebase/mmseg/deploy/segmentation_model.py

```diff
@@ -82,78 +81,50 @@ def forward(self, img: Sequence[torch.Tensor],
-            seg_pred = seg_pred.long().detach().cpu().numpy()
-        # remove unnecessary dim
-        seg_pred = seg_pred.squeeze(1)
-        seg_pred = list(seg_pred)
-        return seg_pred
-
-    def forward_test(self, imgs: torch.Tensor, *args, **kwargs) -> \
-            List[np.ndarray]:
-        """The interface for forward test.
-
-        Args:
-            imgs (torch.Tensor): Input image(s) in [N x C x H x W] format.
-
-        Returns:
-            List[np.ndarray]: A list of segmentation map.
-        """
-        outputs = self.wrapper({self.input_name: imgs})
-        outputs = self.wrapper.output_to_list(outputs)
-        outputs = [out.detach().cpu().numpy() for out in outputs]
-        return outputs
-
-    def show_result(self,
-                    img: np.ndarray,
-                    result: list,
-                    win_name: str = '',
-                    palette: Optional[np.ndarray] = None,
-                    show: bool = True,
-                    opacity: float = 0.5,
-                    out_file: str = None):
-        """Show predictions of segmentation.
-        Args:
-            img: (np.ndarray): Input image to draw predictions.
-            result (list): A list of predictions.
-            win_name (str): The name of visualization window. Default is ''.
-            palette (np.ndarray): The palette of segmentation map.
-            show (bool): Whether to show plotted image in windows. Defaults to
-                `True`.
-            opacity: (float): Opacity of painted segmentation map.
-                    Defaults to `0.5`.
-            out_file (str): Output image file to save drawn predictions.
-
-        Returns:
-            np.ndarray: Drawn image, only if not `show` or `out_file`.
-        """
-        palette = self.PALETTE if palette is None else palette
-        return BaseSegmentor.show_result(
-            self,
-            img,
-            result,
-            palette=palette,
-            opacity=opacity,
-            show=show,
-            win_name=win_name,
-            out_file=out_file)
+        assert mode == 'predict', \
+            'Backend model only support mode==predict,' f' but get {mode}'
+        if inputs.device != torch.device(self.device):
+            get_root_logger().warning(f'expect input device {self.device}'
+                                      f' but get {inputs.device}.')
+        inputs = inputs.to(self.device)
+        batch_outputs = self.wrapper({self.input_name:
+                                      inputs})[self.output_names[0]]
+        return self.pack_result(batch_outputs, data_samples)
+
+    def pack_result(self, batch_outputs, data_samples):
+        predictions = []
+        for seg_pred, data_sample in zip(batch_outputs, data_samples):
+            # resize seg_pred to original image shape
+            if 'img_path' in data_sample.metainfo:
+                image = mmcv.imread(data_sample.metainfo['img_path'])
```

Comment:
Do not use imread if you just want to get the shape of the image.
Use pillow instead.